vm/src/vm.rs

```diff
@@ -823,6 +841,44 @@ impl VirtualMachine {
                 self.push_block(Block::TryExcept { handler: *handler });
                 None
             }
+            bytecode::Instruction::SetupWith { end } => {
+                let obj = self.pop_value();
+                // Call enter:
+                match self.call_method(obj, "__enter__", vec![]) {
+                    Ok(manager) => {
+                        self.push_block(Block::With {
+                            end: *end,
+                            context_manager: manager.clone(),
```

Comment:
Actually, this is more than a naming issue; I think this works for the case where a context manager returns `self` from `__enter__`, but won't work for other cases (because we end up calling `__exit__` on the return value of `__enter__`, not the context manager itself). For example, the current code blows up on the following (whereas CPython is perfectly happy with it):

```py
ls = []

class ContextManager:
    def __enter__(self):
        print('Entrada')
        ls.append(1)
        return ls

    def __exit__(self, exc_type, exc_val, exc_tb):
        ls.append(2)
        print('Wiedersehen')


with ContextManager() as c:
    assert c == [1]

assert ls == [1, 2]
```

Reply:
Wow, okay, valid point, I did not know this, but good to know, I'll add a testcase for this.